### PR TITLE
Branch editappointmentcommand

### PR DESCRIPTION
**Pull request by dvdweien**
Add invalid property/appointment index to messages
Add editappointment support for addressbookparser
Update model to support editappointmentcommand
Update modelmanager to support editappointmentcommand
Update stubs in addappointmentcommandtest and
addpropertycommandtest
Add editappointmentcommand
Add editappointmentcommandparser

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditAppointmentCommand.java [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    public static final String MESSAGE_SUCCESS = "Edited appointment: %1$s";
+    public static final String MESSAGE_DUPLICATE_PROPERTY = "This appointment already exists in the app";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+
+    private final Index index;
+    private final EditAppointmentDescriptor editAppointmentDescriptor;
+
+    /**
+     * Creates an EditPropertyCommand to edit the specified {@code Property}.
+     */
```
Comment (w2vgd) [inline src/main/java/seedu/address/logic/commands/EditAppointmentCommand.java]: Rmb to change this javadoc

File: src/main/java/seedu/address/logic/commands/EditAppointmentCommand.java [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+            + PREFIX_DATE + "DATE "
+            + PREFIX_TIME + "TIME \n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_NAME + "Meet Alex "
+            + PREFIX_REMARK + "at M hotel "
+            + PREFIX_DATE + "17-2-2021 "
+            + PREFIX_TIME + "1500 ";
+
+    public static final String MESSAGE_SUCCESS = "Edited appointment: %1$s";
+    public static final String MESSAGE_DUPLICATE_PROPERTY = "This appointment already exists in the app";
```
Comment (w2vgd) [inline src/main/java/seedu/address/logic/commands/EditAppointmentCommand.java]: Rmb to update this variable name

File: src/main/java/seedu/address/logic/commands/EditAppointmentCommand.java [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+        if (index.getZeroBased() >= model.getAppointmentSize()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_APPOINTMENT_DISPLAYED_INDEX);
+        }
+
+        Appointment appointmentToEdit = model.getAppointment(index.getZeroBased());
+        Appointment editedAppointment = createEditedAppointment(appointmentToEdit, editAppointmentDescriptor);
+
+        if (!appointmentToEdit.isSameAppointment(editedAppointment) && model.hasAppointment(editedAppointment)) {
+            throw new CommandException(MESSAGE_DUPLICATE_PROPERTY);
```
Comment (w2vgd) [inline src/main/java/seedu/address/logic/commands/EditAppointmentCommand.java]: Rmb to update this variable name

File: src/main/java/seedu/address/logic/commands/EditAppointmentCommand.java [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+        }
+
+        model.setAppointment(appointmentToEdit, editedAppointment);
+        return new CommandResult(String.format(MESSAGE_SUCCESS, editedAppointment));
+    }
+
+    /**
+     * Creates and returns a {@code Property} with the details of {@code propertyToEdit}
+     * edited with {@code editPropertyDescriptor}.
+     */
```
Comment (w2vgd) [inline src/main/java/seedu/address/logic/commands/EditAppointmentCommand.java]: Rmb to update this javadoc

File: src/main/java/seedu/address/logic/parser/EditAppointmentCommandParser.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.EditAppointmentCommand;
+import seedu.address.logic.commands.EditAppointmentCommand.EditAppointmentDescriptor;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+
+/**
+ * Parses input arguments and creates a new EditPropertyCommand object.
+ */
```
Comment (w2vgd) [inline src/main/java/seedu/address/logic/parser/EditAppointmentCommandParser.java]: Rmb to update this javadoc

**Review by w2vgd — COMMENTED** [review]
LGTM with a few comments

File: src/main/java/seedu/address/logic/commands/EditAppointmentCommand.java (hunk 1 above)
Comment (dvdweien) [reply src/main/java/seedu/address/logic/commands/EditAppointmentCommand.java]: Thanks for catching it! should all be fixed now

**Review by w2vgd — APPROVED** [review]
LGTM

**Pull request merged**